File: app/src/main/java/org/kindone/willingtodo/pomodorotimer/TimerPaused.java
package org.kindone.willingtodo.pomodorotimer;

/**
 * Created by kindone on 2017. 1. 1..
 */

public class TimerPaused extends TimerState {
    final private long mRemainingMs;

    public TimerPaused(TimerContext context, long remainingMs) {
        super(context);
        mRemainingMs = remainingMs;
    }

    @Override
    public boolean start() {
        resume();
        return CHANGED;
    }

    @Override
    public boolean pause() {
        // DO NOTHING
        return UNCHANGED;
    }

    @Override
    public boolean resume() {
        changeState(new TimerRunning(mContext, mRemainingMs));
        return CHANGED;
    }

    @Override
    public boolean stop() {
        changeState(new TimerStopped(mContext));
        return UNCHANGED;
    }

    @Override
    public long getRemainingTimeMs() {
        return mRemainingMs > 0 ? mRemainingMs : 0;
    }

    public boolean isRunning() {
        return false;
    }

    public boolean isStopped() {
        return false;
    }
}

File: app/src/main/java/org/kindone/willingtodo/pomodorotimer/TimerPaused.kt
package org.kindone.willingtodo.pomodorotimer

/**
 * Created by kindone on 2017. 1. 1..
 */

class TimerPaused(context: TimerContext, private val mRemainingMs: Long) : TimerState(context) {


    override fun start(): Boolean {
        resume()
        return TimerState.CHANGED
    }

    override fun pause(): Boolean {
        // DO NOTHING
        return TimerState.UNCHANGED
    }

    override fun resume(): Boolean {
        changeState(TimerRunning(mContext, mRemainingMs))
        return TimerState.CHANGED
    }

    override fun stop(): Boolean {
        changeState(TimerStopped(mContext))
        return TimerState.UNCHANGED
    }

    override fun getRemainingTimeMs(): Long {
        return if (mRemainingMs > 0) mRemainingMs else 0
    }

    override fun getStartedTimeMs(): Long {
        return -1L
    }

    override fun isRunning(): Boolean {
        return false
    }

    override fun isStopped(): Boolean {
        return false
    }
}
